chain/vm/runtime.go

```diff
@@ -109,6 +108,9 @@ func (rs *Runtime) Get(c cid.Cid, o vmr.CBORUnmarshaler) bool {
 	if err := rs.cst.Get(context.TODO(), c, o); err != nil {
 		var nfe notFoundErr
 		if xerrors.As(err, &nfe) && nfe.IsNotFound() {
+			if xerrors.As(err, new(cbor.SerializationError)) {
```

Comment:
I'm not super familiar with xerrors yet, but is `As()` a sufficient test? One reading of the xerrors documentation would suggest that any error that has is compatible with `cbor.SerializationError` would match this, even if not being exactly the same type. And if the `cbor.IpldStore` interface in `rs.cst` is satisfied by something other than `BasicIpldStore` (e.g. a read/write caching wrapper), it might throw some different error type?

 E.g. the error thrown in the test is not actually a `cbor.SerializationError`, though you are using a vanilla store which probably wraps it to make it so. There is no test that if `Get()` returns some _other_ error type that it's _not_ translated to ErrSerialization.

FYI A similar change in go-filecoin : https://github.com/filecoin-project/go-filecoin/pull/4060